test/test_binary_ufuncs.py

```diff
@@ -2046,18 +2046,18 @@ def test_heaviside(self, device, dtypes):
 
     @onlyCUDA
     def test_heaviside_cross_device(self, device):
-        x = torch.tensor([-9, 5, 0, 6, -2, 2], device='cuda')
-        y = torch.tensor(0)
```

Comment:
It's seem that heaviside allows `cuda Tensor` x `cpu Tensor` in old way, and it's not supported in structured kernel. I thought this block is reasonable, since the error msg says `torch.heaviside expect all tensors to be on the same device`, so I fixed the test case.

Reply:
Nyeh, this is a regression. In fact you're the one who originally fixed this, right, in  https://github.com/pytorch/pytorch/pull/46831 ?

CPU scalar should keep working. At a guess, you should add

```
  device_check: NoCheck   # TensorIterator
```

to all of the native functions entries for heaviside. Then the old logic should work.

Reply:
It works! Thank you so much for the guidance！